api/v1beta1/azurecluster_validation_test.go

```diff
@@ -1082,12 +1221,67 @@ func TestValidateAPIServerLB(t *testing.T) {
 			cpCIDRS: []string{"10.0.0.0/24", "10.1.0.0/24"},
 			wantErr: false,
 		},
+		{
+			name:        "public LB with in-range private IP with feature flag APIServerILB enabled",
+			featureGate: feature.APIServerILB,
+			lb: LoadBalancerSpec{
+				FrontendIPs: []FrontendIP{
+					{
+						Name: "ip-1",
+						FrontendIPClass: FrontendIPClass{
+							PrivateIPAddress: "10.0.0.123",
+						},
+					},
+					{
+						Name: "ip-2",
+						PublicIP: &PublicIPSpec{
+							Name:    "my-valid-ip",
+							DNSName: "my-valid-ip",
+						},
+					},
+				},
+				LoadBalancerClassSpec: LoadBalancerClassSpec{
+					Type: Public,
+					SKU:  SKUStandard,
+				},
+				Name: "my-private-lb",
+			},
+			cpCIDRS: []string{"10.0.0.0/24"},
+			wantErr: false,
+		},
+		{
+			name:        "public LB with out of range private IP with feature flag APIServerILB enabled",
+			featureGate: feature.APIServerILB,
+			lb: LoadBalancerSpec{
+				FrontendIPs: []FrontendIP{
+					{
+						Name: "ip-1",
+						FrontendIPClass: FrontendIPClass{
+							PrivateIPAddress: "20.1.2.3",
+						},
+					},
+				},
+				LoadBalancerClassSpec: LoadBalancerClassSpec{
+					Type: Public,
+				},
+			},
+			cpCIDRS: []string{"10.0.0.0/24", "10.1.0.0/24"},
+			wantErr: true,
+			expectedErr: field.Error{
+				Type:     "FieldValueInvalid",
+				Field:    "apiServerLB.frontendIPConfigs[0].privateIP",
+				BadValue: "20.1.2.3",
+				Detail:   "Internal LB IP address needs to be in control plane subnet range ([10.0.0.0/24 10.1.0.0/24])",
+			},
+		},
 	}
 
 	for _, test := range testcases {
 		t.Run(test.name, func(t *testing.T) {
-			t.Parallel()
```

Comment:
Was this meant to be removed?

Reply:
Yep, `t.Parallel()` was meant to be removed when working with global feature flag. Ref: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e0f2b44aec58ed04bbd754a98bace470bea32fff/api/v1beta1/azurecluster_validation_test.go#L1282-L1284

Having `t.Parallel()` when setting global values will mess with the test leading to erroneous output or race conditions( race condition not applicable in our scenario). 
Hence, I had to remove `t.Parallel()`.